Framework/DataHandling/src/LoadMuonNexusV2.cpp

```diff
@@ -0,0 +1,227 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidDataHandling/LoadMuonNexusV2.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/RegisterFileLoader.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidDataHandling/LoadISISNexus2.h"
+#include "MantidDataHandling/LoadMuonNexusV2Helper.h"
+#include "MantidDataHandling/SinglePeriodLoadMuonStrategy.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/ListValidator.h"
+
+#include <vector>
+
+namespace Mantid {
+namespace DataHandling {
+
+DECLARE_NEXUS_HDF5_FILELOADER_ALGORITHM(LoadMuonNexusV2)
+
+using namespace Kernel;
+using namespace API;
+using namespace NeXus;
+using namespace HistogramData;
+using std::size_t;
+using namespace DataObjects;
+
+/// Empty default constructor
+LoadMuonNexusV2::LoadMuonNexusV2()
+    : m_filename(), m_entrynumber(0), m_isFileMultiPeriod(false),
+      m_multiPeriodsLoaded(false) {}
+
+/**
+ * Return the confidence criteria for this algorithm can load the file
+ * @param descriptor A descriptor for the file
+ * @returns An integer specifying the confidence level. 0 indicates it will not
+ * be used
+ */
+int LoadMuonNexusV2::confidence(NexusHDF5Descriptor &descriptor) const {
+  // Without this entry we cannot use LoadISISNexus
+  if (!descriptor.isEntry("/raw_data_1", "NXentry")) {
```

Comment:
Can we define all of the constant strings once to avoid repetition. A bit like [IndexPeaks](https://github.com/mantidproject/mantid/blob/master/Framework/Crystal/src/IndexPeaks.cpp#L39).

Reply:
Yeah good point, I've extracted all the NeXus entry related strings into a separate namespace containing the string definition, as in IndexPeaks.